### PR TITLE
[mod] PR template: "Before You Submit" add hint about CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by return42**
## What does this PR do?

Extend PR template by a new section "Before You Submit" and add hint about [CONTRIBUTING.md](https://github.com/searxng/searxng/blob/master/CONTRIBUTING.md)

## Why is this change important?

Contributors should be able to take note of our rules ([CONTRIBUTING.md](https://github.com/searxng/searxng/blob/master/CONTRIBUTING.md))

## How to test this PR locally?

HTML rendered preview: [PULL_REQUEST_TEMPLATE.md](
https://github.com/return42/searxng/blob/mod-PR-template/PULL_REQUEST_TEMPLATE.md)

## Before You Submit
<small>Please ensure you check the following items to indicate that you've read this section and completed each task</small>
- [x] My code follows the [Contribution Guidelines](https://github.com/searxng/searxng/blob/master/CONTRIBUTING.md)
- [x] I give expressed consent for my work to be used in this repo
- [x] I have tested my work and it functions as intended
- [x] I have included documentation if the change requires such
If you have problems, e.g. with the documentation (reST markup), then we can help in the review process.

### CODE REVIEW

**Review by Bnyro — COMMENTED** [review]
1. [CONTRIBUTING.md](https://github.com/searxng/searxng/blob/master/CONTRIBUTING.md) doesn't mention the `[feat] engineXY: do XY` commit style, should we add that to `CONTRIBUTING.md`?
2. Should we ask contributors to squash their changes (most new contributors open PRs with lots of commits), or would that just overcomplicate things for new contributors?
3. `I give expressed consent for my work to be used in this repo`: Why else would somebody open a pull request? Is this for some legal reason? Did SearXNG ever face problems with that topic?
4. Instead of `I give expressed consent for my work to be used in this repo` I would maybe do something like `I wrote this code myself and neither copied parts of it from other projects with restrictive licenses or used generative AI to create the code.`

**Comment by return42** [comment]
> most new contributors open PRs with lots of commits ..

Yes, and in the `master` branch .. even it might be complicated for one or the other, we should add it to the rules .. / those who should pay attention to it will not read it anyway ;-)

**Comment by glanham-jr** [comment]
Should unit tests be leveraged as a soft requirement checkbox? Maybe something like "I've written unit tests, if any, to the best of my ability".

**Comment by Bnyro** [comment]
> Should unit tests be leveraged as a soft requirement checkbox? Maybe something like "I've written unit tests, if any, to the best of my ability".
>
I think that makes sense if the code change is about SearXNG in general (e.g. webapp.py) or engine utility functions, but it probably doesn't make sense to write exhaustive unit tests for each added / modified engine. Apart from that we already have general unit tests for testing engines. So to avoid confusion, I would rephrase it to something like
`I've written unit tests for all added code that's not engine-specific to the best of my ability.`

Though I'm not too sure if not having written unit tests should prevent somebody from opening a pull request, it should likely only be an optional requirement.

**Comment by return42** [comment]
Superseded by https://github.com/searxng/searxng/pull/5973
